core/modules/cart/store/actions.ts

```diff
@@ -674,6 +674,16 @@ const actions: ActionTree<CartState, RootState> = {
         console.error(event.result)
       }
     }
+    // Set isVirtualCart to true if downloadable and virtual products in cart
+    const clientItems = rootStore.state.cart.cartItems
+    const isVirtualCart = clientItems.every((itm) => {
+      return itm.type_id === 'downloadable' || itm.type_id === 'virtual'
+    })
+    rootStore.state.cart.isVirtualCart = false
+    if (isVirtualCart) {
+      rootStore.state.cart.isVirtualCart = true
```

Comment:
Please do use mutation either Vue.set() and use the state from “context” not from the rootState

Reply:
Regarding our naming convention 
https://github.com/DivanteLtd/vue-storefront/issues/2069 we should have “state.cartIsVirtual” property and then getter “getters.isVirtualCart”. You should use getter and “mapGetters” from Vue instead of computed properties to access this variable in components

Reply:
Please also add the cartIsVirtual property to the state type definition:
https://github.com/DivanteLtd/vue-storefront/blob/master/core/modules/cart/types/CartState.ts